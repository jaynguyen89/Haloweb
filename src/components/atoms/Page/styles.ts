import vars from 'src/commons/variables/cssVariables.scss';
import { SxProps, Theme } from '@mui/material';

export const pageSx: SxProps<Theme> = {
    width: '100%',
    height: '100%',
    margin: vars.none,
    padding: vars.none,
};

export const containerSx: SxProps<Theme> = {
    mt: vars.medium,
    mb: vars.xxsmall,
    padding: vars.none,
};
