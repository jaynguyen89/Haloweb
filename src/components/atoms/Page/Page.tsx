import React, { FunctionComponent, ReactNode } from 'react';
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import { pageSx, containerSx } from 'src/components/atoms/Page/styles';

const Page: FunctionComponent<{
    children: ReactNode,
    pageStyle?: React.CSSProperties,
    pageClassName?: string,
    containerStyle?: React.CSSProperties,
    containerClassName?: string,
}> = ({
    children,
    pageStyle,
    pageClassName,
    containerStyle,
    containerClassName,
}) => {
    return (
        <Box sx={pageSx} style={pageStyle} className={pageClassName}>
            <Container
                maxWidth='xl'
                sx={containerSx}
                style={containerStyle}
                className={containerClassName}
            >
                {children}
            </Container>
        </Box>
    );
};

export default Page;
