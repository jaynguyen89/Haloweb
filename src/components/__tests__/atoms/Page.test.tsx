import React from 'react';
import { render, queryByAttribute } from '@testing-library/react'
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import Page from 'src/components/atoms/Page/Page';

configure({ adapter: new Adapter() });
jest.mock('src/commons/variables/cssVariables.scss', () => ({
    none: 0,
    xxsmall: 1,
    medium: 2,
}));

describe('Page', () => {
    const props = {
        pageClassName: 'page-class-name',
        containerClassName: 'container-class-name',
        pageStyle: { margin: '20px' },
        containerStyle: { padding: '30px' },
    };

    it('renders successfully', () => {
        const wrapper = shallow(
            <Page {...props}><span id='span-id'>Test Child</span></Page>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render the given child', () => {
        const wrapper = render(
            <Page {...props}>
                <span id='span-id'>Test Child</span>
            </Page>
        );

        expect(wrapper.container.getElementsByTagName('span').length).toBe(1);

        const getById = queryByAttribute.bind(null, 'id');
        expect(getById(wrapper.container, 'span-id')).toBeDefined();

        expect(wrapper.container.getElementsByClassName('page-class-name').length).toBe(1);
        expect(wrapper.container.getElementsByClassName('container-class-name').length).toBe(1);
    });
});
