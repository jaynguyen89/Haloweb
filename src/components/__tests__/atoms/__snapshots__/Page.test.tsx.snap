// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders successfully 1`] = `
<ForwardRef(Box)
  className="page-class-name"
  style={
    Object {
      "margin": "20px",
    }
  }
  sx={
    Object {
      "height": "100%",
      "margin": 0,
      "padding": 0,
      "width": "100%",
    }
  }
>
  <ForwardRef(Container)
    className="container-class-name"
    maxWidth="xl"
    style={
      Object {
        "padding": "30px",
      }
    }
    sx={
      Object {
        "mb": 1,
        "mt": 2,
        "padding": 0,
      }
    }
  >
    <span
      id="span-id"
    >
      Test Child
    </span>
  </ForwardRef(Container)>
</ForwardRef(Box)>
`;
